build.yaml

```diff
@@ -612,35 +711,37 @@ common_cli:
   # this is a wrapper that we can use for building a script that will be
   # populated with different cli actions.
   # we copy the resulting build.sh file to the vagrant box for execution.
-  begin_build_sh_EOF: &begin_build_sh_EOF cat <<'EOF' > build.sh
+  begin_build_sh_EOF: &begin_build_sh_EOF cat <<'EOF_BUILD_SH' > build.sh
 
   end_build_sh_EOF: &end_build_sh_EOF |
-     EOF
+     EOF_BUILD_SH
      chmod 755 build.sh
 
-  copy_files_to_osx_yosemite_box: &copy_files_to_osx_yosemite_box |
-    # this sets some variables used by the tests, since the tests run inside
-    # virtualbox we store them to a file and copy them to the vagrant box.
-    echo "export GIT_COMMIT=\${GIT_COMMIT}" > extra-vars.sh
-    chmod 755 extra-vars.sh
-    # gather the connection string for this vagrantbox
+  run_build_sh_script_on_vagrant_box: &run_build_sh_script_on_vagrant_box |
+    # copy project files to the vagrant box
     vagrant ssh-config > ssh-config
     rsync -ave 'ssh -F ssh-config' . default:.
-    vagrant scp /tmp/pip.sh /tmp/pip.sh
-    vagrant scp build.sh build.sh
-    vagrant ssh -c 'chmod 755 /tmp/pip.sh'
-    vagrant ssh -c 'chmod 755 build.sh'
 
-  run_build_sh_script_on_vagrant_box: &run_build_sh_script_on_vagrant_box |
+    # this sets some variables used by the tests, since the tests run inside
+    # virtualbox we store them to a file and copy them to the vagrant box.
+    echo "export GIT_COMMIT=\${GIT_COMMIT}" > git-commit.sh
+    for item in /tmp/pip.sh build.sh git-commit.sh
+    do
+      vagrant scp \${item} default:\${item}
+      vagrant ssh -c "chmod 755 \${item}"
+    done
+
     # make sure we don't abort on the first error
     set +e
     # run the build.sh script inside our vagrant box
-    vagrant ssh -c 'bash build.sh'
-    JOB_EXIT_STATUS="\$( updateExitStatus \$? )"
+    vagrant ssh -c 'bash build.sh' || abort_build
+
+    # this removes the secrets files we used during provisioning
```

Comment:
sorry, which step ?
github excerpts can be confusing, are you refering to L737 ?

if the build.sh fails, then abort_build is executed
'||' in bash means 'OR' and evaluated left to righ

```
\>ls /non-existing-file || echo "that's was borked"
ls: cannot access /non-existing-file: No such file or directory
that's was borked
```